src/hoc/withEditing/withEditing.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+import {StickyFooter} from '../../components/stickyFooter/StickyFooter';
+import * as styles from '../../components/stickyFooter/StickyFooter.scss';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {IDispatch, ReduxConnect} from '../../utils/ReduxUtils';
+import {toggleDirtyComponent} from './withEditingActions';
+import {getIsDirty} from './withEditingReducers';
+
+export interface IWithEditing {
+    id: string;
+    isDirty?: boolean;
+    footerChildren?: React.ReactNode;
+}
+
+export interface IWithEditStateProps {
+    isDirty: boolean;
+}
+
+export interface IWithEditDispatchProps {
+    toggleDirtyComponent: (isDirty: boolean) => void;
+}
+
+export const withEditing = (config: IWithEditing) => (Component: React.ComponentClass<any, any>): React.ComponentClass<any, any> => {
+    const mapStateToProps = (state: IReactVaporState): IWithEditStateProps => ({
+        isDirty: getIsDirty(state, config.id),
+    });
+
+    const mapDispatchToProps = (dispatch: IDispatch): IWithEditDispatchProps => ({
+        toggleDirtyComponent: (isDirty: boolean) => dispatch(toggleDirtyComponent(config.id, isDirty)),
+    });
+
+    @ReduxConnect(mapStateToProps, mapDispatchToProps)
+    class ComponentWithEditing extends React.Component<any, any> {
+        componentWillUnmount() {
+            callIfDefined(this.props.toggleDirtyComponent, false);
+        }
+
+        render() {
+            return (
+                <div className={styles.stickyFooterPadding}>
+                    <Component {..._.omit(this.props, 'isDirty', 'footerChildren')}>

```

Comment:
I think we need an official statement. ttypescript with the keys function or not?

Reply:
most likely yes i just thought we wanted to move away from it

Reply:
i actually don't need to omit anything :)